x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Percentile.java

```diff
@@ -16,31 +17,44 @@
 import org.elasticsearch.xpack.sql.type.SqlDataTypeConverter;
 
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-import static java.util.Collections.singletonList;
 import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isFoldable;
 import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isNumeric;
+import static org.elasticsearch.xpack.ql.expression.function.Functions.countOfNonNullOptionalArgs;
+import static org.elasticsearch.xpack.sql.expression.function.aggregate.PercentileMethodConfiguration.defaultMethod;
+import static org.elasticsearch.xpack.sql.expression.function.aggregate.PercentileMethodConfiguration.defaultMethodParameter;
+import static org.elasticsearch.xpack.sql.expression.function.aggregate.PercentileMethodConfiguration.resolvePercentileConfiguration;
 
-public class Percentile extends NumericAggregate implements EnclosedAgg {
+public class Percentile extends NumericAggregate implements EnclosedAgg, TwoOptionalArguments {
 
     private final Expression percent;
+    private final Expression method;
+    private final Expression methodParameter;
 
-    public Percentile(Source source, Expression field, Expression percent) {
-        super(source, field, singletonList(percent));
+    public Percentile(Source source, Expression field, Expression percent, Expression method, Expression methodParameter) {
+        super(source, field, Stream.of(percent, (method = defaultMethod(source, method)),
+            (methodParameter = defaultMethodParameter(methodParameter))).filter(Objects::nonNull).collect(Collectors.toList()));
```

Comment:
Instead of an in-line stream creation/filtering/collection inside a super invocation (a no-no), extract this into a static utility method.